src/rpc/net.cpp

```diff
@@ -543,17 +543,16 @@ UniValue setban(const UniValue& params, bool fHelp)
 
     CSubNet subNet;
     CNetAddr netAddr;
-    bool isSubnet = false;
+    bool isSubnet, isValid;
 
-    if (params[0].get_str().find("/") != string::npos)
-        isSubnet = true;
+    isSubnet = params[0].get_str().find("/") != string::npos;
 
-    if (!isSubnet)
-        netAddr = CNetAddr(params[0].get_str());
+    if (isSubnet)
+        isValid = LookupSubNet(params[0].get_str().c_str(), subNet);
     else
-        subNet = CSubNet(params[0].get_str());
+        isValid = LookupHost(params[0].get_str().c_str(), netAddr, false);
```

Comment:
People have argued about how to return both an object and an error code for decades and also argued about how to programatically enforce the checking of the error code.  This is why exceptions were invented.  

I feel that changing this code from one return style to another is simply expressing your personal preference.  This is an example of unnecessary change -- code "churn".

Should these objects be converted to RAII and throw an exception?  That would address the problem of not checking returned error codes.  However, does the result in these lightly used classes justify the effort to switch to exceptions, to undergo a careful analysis of exception handling whereever CNetAddr and CSubNet are created, and the re-education of all our developers about CNetAddr and CSubNet?  My intuition says no...